launcher/ui/pages/global/JavaPage.cpp

```diff
@@ -95,7 +95,7 @@ void JavaPage::applySettings()
 
     // Java Settings
     s->set("JavaPath", ui->javaPathTextBox->text());
-    s->set("JvmArgs", ui->jvmArgsTextBox->text());
+    s->set("JvmArgs", ui->jvmArgsTextBox->toPlainText());
```

Comment:
```suggestion
    s->set("JvmArgs", ui->jvmArgsTextBox->toPlainText().replace("\n", " "));
```

The behavior when a user saves arguments with new lines differs between the global setting and the instance setting.

In the instance setting (already multi-line), when a user enters newlines into the JVM arguments box and saves their settings, the new lines are automatically converted to spaces and saved. For reference, it does this:

https://github.com/PolyMC/PolyMC/blob/699ad316f0d90580fa13d570d6c25aff903a470d/launcher/ui/pages/instance/InstanceSettingsPage.cpp#L195

In this new multi-line global settings text box, if the user types new lines, the new lines are saved. If a user saves arguments with new lines, the game crashes on startup because it doesn't recognize that the new lines separate different arguments (at least on my system):

```
Unrecognized VM option 'UseG1GC
-Dsun.rmi.dgc.server.gcInterval=2147483646
-XX:+UnlockExperimentalVMOptions
-XX:G1NewSizePercent=20
-XX:G1ReservePercent=20
-XX:MaxGCPauseMillis=50
-XX:G1HeapRegionSize=32M'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Minecraft process ID: 86444


Process exited with code 1.
```

tl;dr: the behavior when users enter a new line into the arguments box should be standard between the global and instance settings. The easiest way to do this would just be to convert new lines to spaces here.